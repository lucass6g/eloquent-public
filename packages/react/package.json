{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
