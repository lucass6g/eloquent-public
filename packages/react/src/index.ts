export {Switch, Case, Default} from './components/conditional-switch'