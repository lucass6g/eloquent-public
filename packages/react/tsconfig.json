{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
