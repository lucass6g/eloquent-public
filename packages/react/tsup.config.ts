import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})