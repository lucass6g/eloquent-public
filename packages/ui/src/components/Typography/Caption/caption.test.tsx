import { render, screen } from "@testing-library/react"
import {describe, it, expect} from "vitest"
import { Caption } from "."

describe("Caption Component", () => {
    it("should rendering the component correctly", () => {
        render(<Caption size="default">O Brasil é caramelo</Caption>);
        expect(screen.getByText("O Brasil é caramelo")).toBeInTheDocument() 
    })

    it("should render component with different sizes", () => {
       const { rerender } = render(<Caption size="default">Size default</Caption>);
        let sizeElement = screen.getByText("Size default");
        expect(sizeElement).toHaveClass("text-sm");
         
        rerender(<Caption size="sm">Size sm</Caption>);
        sizeElement = screen.getByText("Size sm");
        expect(sizeElement).toHaveClass("text-xs");
        
    })
})