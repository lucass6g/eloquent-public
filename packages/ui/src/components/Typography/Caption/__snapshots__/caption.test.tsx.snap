// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Caption > should match the snapshot 1`] = `
<body>
  <div>
    <h6
      class="text-foreground font-normal text-sm"
    />
  </div>
</body>
`;
